src/hooks/pg.test.js

```diff
@@ -0,0 +1,175 @@
+    const client = createHookedPgClient({ error });
+    let errorIsRaised = false;
+
+    try {
+      await client.query('SELECT * from users');
+    } catch (e) {
+      expect(e.name).toEqual('Error');
+      expect(e.message).toEqual('RandomError');
+      errorIsRaised = true;
+    }
+
+    const spans = SpansContainer.getSpans();
+    expect(spans).toEqual([
+      createBaseBuilderFromSpan(spans[0])
+        .withQuery('SELECT * from users')
+        .withConnectionParameters(DUMMY_PG_OPTIONS)
+        .withError(payloadStringify(error))
+        .build(),
+    ]);
+    expect(errorIsRaised).toBeTruthy();
+  });
+
+  test('hookPg -> query (text: string) => Promise -> error (with catch)', async () => {
+    const error = new Error('RandomError');
+    const client = createHookedPgClient({ error });
+    let errorIsRaised = false;
+
+    await client.query('SELECT * from users').catch(e => {
+      expect(e.name).toEqual('Error');
+      expect(e.message).toEqual('RandomError');
+      errorIsRaised = true;
+    });
+
+    const spans = SpansContainer.getSpans();
+    expect(spans).toEqual([
+      createBaseBuilderFromSpan(spans[0])
+        .withQuery('SELECT * from users')
+        .withConnectionParameters(DUMMY_PG_OPTIONS)
+        .withError(payloadStringify(error))
+        .build(),
+    ]);
+    expect(errorIsRaised).toBeTruthy();
+  });
+
+  test('hookPg -> query (text: string, values: Array) => Promise -> success', async () => {
+    const client = createHookedPgClient();
+
+    await client.query('SELECT * from users', ['Value']);
+
+    const spans = SpansContainer.getSpans();
+    expect(spans).toEqual([
+      createBaseBuilderFromSpan(spans[0])
+        .withQuery('SELECT * from users')
+        .withConnectionParameters(DUMMY_PG_OPTIONS)
+        .withResponse(createExpectedResponse())
+        .build(),
+    ]);
+  });
+
+  test('hookPg -> query (text: string, callback: Function) => void -> success', done => {
+    const client = createHookedPgClient();
+
+    const testFunc = () => {
+      const spans = SpansContainer.getSpans();
+      expect(spans).toEqual([
+        createBaseBuilderFromSpan(spans[0])
+          .withQuery('SELECT * from users')
+          .withConnectionParameters(DUMMY_PG_OPTIONS)
+          .withResponse(createExpectedResponse())
+          .build(),
+      ]);
+      done();
+    };
+
+    client.query('SELECT * from users', testFunc);
+  });
+
+  test('hookPg -> query (text: string, callback: Function) => void -> error', done => {
+    const error = new Error('RandomError');
+    const client = createHookedPgClient({ error });
+
+    const testFunc = err => {
+      const spans = SpansContainer.getSpans();
+      expect(spans).toEqual([
+        createBaseBuilderFromSpan(spans[0])
+          .withQuery('SELECT * from users')
+          .withConnectionParameters(DUMMY_PG_OPTIONS)
+          .withError(payloadStringify(error))
+          .build(),
+      ]);
+      expect(err).toEqual(err);
```

Comment:
What do we test here?

Reply:
That the user will get the exception (that we not effecting this, because we catch and re-raise it)